### PR TITLE
[WFLY-19152] Add indices of APIs annotated with annotations indicating unstable API

### PR DESCRIPTION
**Pull request by kabir**
https://issues.redhat.com/browse/WFLY-19152

Needs https://github.com/wildfly/wildfly-core/pull/5913 (this also contains the link to the analysis document)

Also requires https://github.com/wildfly/wildfly/pull/17849 for the reload to a lower stability level

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-13682](https://issues.redhat.com/browse/WFLY-13682)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

### CODE REVIEW

**Review by ehsavoie — APPROVED** [review]

**Review by ehsavoie — APPROVED** [review]

**Review by pferraro — APPROVED** [review]
This all looks good to me - though I made several suggestions on the WFCORE pull request itself.

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Core_Management.adoc [hunk 1]
```diff
@@ -132,3 +132,37 @@ Now we can list the last configuration changes :
     }]
 }
 ----
+
+[[unstable-api-annotation]]
+== [Preview Feature] Reporting of API elements annotated with annotations indicating unstable API
+
+Note: This feature only exists at the `preview` stability level or lower.
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Core_Management.adoc:139]: Please make part of this a link to https://docs.wildfly.org/32/Admin_Guide.html#Feature_stability_levels

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Core_Management.adoc [hunk 2]
```diff
@@ -132,3 +132,37 @@ Now we can list the last configuration changes :
...
+
+Some libraries use annotations on classes/interfaces/members to indicate that the API is considered to be unstable, and likely to change or go away in a future release. For example Hibernate uses the `org.hibernate.Incubating` annotation to mark such API elements.
+
+WildFly can scan for usage of such API when your applications are deployed into WildFly, but it needs to be enabled via addition of the `/subsystem=core-management/service=unstable-api-annotations` resource. Once enabled, you can choose whether usage of these API elements should result in a warning in the logs, or a failed deployment. The next two child sections show examples of how to configure the scanner.
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Core_Management.adoc:143]: s/for usage/for the use/g
s/usage/the use/g

Perhaps this is my personal preference, as using 'usage' as you have here is AFAICT technically correct per at least some definitions I find. But it always feels off to me, like it's an atypical way to use that word.

**Pull request merged**